node/auth/authenticator.go

```diff
@@ -120,7 +121,7 @@ func (a *requestAuthenticator) AuthenticateStoreChunksRequest(
 
 	key, err := a.getDisperserKey(ctx, now, request.DisperserID)
 	if err != nil {
-		return fmt.Errorf("failed to get operator key: %w", err)
+		return api.WrapGRPCError(err, "failed to get operator key")
```

Comment:
do we know for sure that err here is a grpc error?

Reply:
The point of this wrapper is to make it easier for APIs to return canonical errors, without having to read all underlying code

Reply:
I personally think this is a bad idea though. It's better to be SUPER clear on the semantics on functions. If a function returns a grpc error, then ALL of its returned errors should be GRPC. If not, then the function is too large and should be broken down to respect this invariant. This way error handling semantics and documentation is simple, and you can't go wrong.

Otherwise if "everything is easy" people just don't know what they are dealing with, and to extend a function or return a new error code they need to go read the entire code path to understand what kind of errors they could be receiving in different cases.

Reply:
better to code for easy static checkability

Reply:
the issue is you cannot ask all functions to return grpc error which fits only at interface level